### PR TITLE
Drag and drop file zone for upload questions

### PR DESCRIPTION
**Pull request by hammerfunk**

### CODE REVIEW

File: app/assets/javascripts/application.js.coffee [hunk 1]
```diff
@@ -309,13 +314,46 @@ jQuery ->
     remove_link = $("<a>").addClass("remove-link").prop("href", "#").text("Remove")
     div.append(remove_link)
 
+  isEventSupported = (eventName, element) ->
+    TAGNAMES = {'select': 'input', 'change': 'input', 'submit': 'form', 'reset': 'form', 'error': 'img', 'load': 'img', 'abort': 'img'}
```
Comment (matthewford) [inline app/assets/javascripts/application.js.coffee]: Line exceeds maximum allowed length

File: app/assets/javascripts/application.js.coffee [hunk 2]
```diff
@@ -309,13 +314,46 @@ jQuery ->
...
+
+        if typeof element[eventName] != 'undefined'
+          element[eventName] = undefined
+        element.removeAttribute(eventName)
+
+    element = null
+    return isSupported
+
+  isMobile = ->
+    return /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent)
```
Comment (matthewford) [inline app/assets/javascripts/application.js.coffee]: Line exceeds maximum allowed length

File: app/assets/javascripts/application.js.coffee [hunk 3]
```diff
@@ -309,13 +314,54 @@ jQuery ->
     remove_link = $("<a>").addClass("remove-link").prop("href", "#").text("Remove")
     div.append(remove_link)
 
+  isEventSupported = (eventName, element) ->
+    TAGNAMES = {
+                 'select': 'input',
```
Comment (matthewford) [inline app/assets/javascripts/application.js.coffee]: Line contains inconsistent indentation

File: app/assets/javascripts/application.js.coffee [hunk 4]
```diff
@@ -309,13 +314,52 @@ jQuery ->
     remove_link = $("<a>").addClass("remove-link").prop("href", "#").text("Remove")
     div.append(remove_link)
 
+  isEventSupported = (eventName, element) ->
+    TAGNAMES = {'select': 'input',
+                'change': 'input',
```
Comment (matthewford) [inline app/assets/javascripts/application.js.coffee]: [stdin]:319:25: error: unexpected :
                'change': 'input',
                        ^

**Comment by dreamfall** [comment]
looks good, waiting for travis

**Pull request merged**